[devex] Update hotfix release command to work with new automated backporting process

christian-byrne opened the pull request:
## Summary

Updates the `/create-hotfix-release` Claude command to align with modern automated backport workflows and adds complete ComfyUI integration automation.

## What's Changed

### 🔧 Workflow Improvements
- **Smart backport detection** - Checks for existing automated backport PRs and their status
- **Intelligent path routing** - Skip to version bump if backports already merged
- **Guided user actions** - Clear instructions for pending backport PRs that need merging
- **Fallback to manual** - Cherry-picking only when automation fails

### 🤖 Automation Enhancements  
- **Fork management** - Automatic fork creation and reuse for ComfyUI PRs
- **Requirements.txt updates** - Complete automation for ComfyUI integration
- **Draft release publishing** - Critical step to uncheck "Set as latest release"
- **Duplicate prevention** - Backport workflow now prevents double triggers

### 📋 Process Organization
- **Clear process overview** - 6-step workflow with smart path detection
- **Context-aware documentation** - Explains when to use each workflow path
- **Updated step numbering** - Consistent cross-references throughout

### 🔄 Backport Workflow Updates
- **Label trigger support** - Workflow now triggers when `needs-backport` labels added to merged PRs
- **Idempotency protection** - Prevents duplicate backport PRs from double triggers
- **Existing backport detection** - Skips work if backport PRs already exist

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-5271-devex-Update-hotfix-release-command-for-modern-automated-backport-workflow-25f6d73d365081baa778d792bc525f55) by [Unito](https://www.unito.io)

christian-byrne committed:
Update hotfix release command for modern automated backport workflow

- Add Step 0 to check automated backport status first
- Emphasize this command is fallback when automation fails
- Add critical draft release publishing step (uncheck 'Set as latest')
- Add ComfyUI requirements.txt PR creation with exact template
- Update workflow context for modern automated backports

christian-byrne committed:
Restructure hotfix release command for modern workflow

- Add clear process overview and context at top
- Step 1: Try automated backports first (via labels)
- Fallback to manual cherry-picking only if automation fails
- Add critical draft release publishing step (uncheck 'Set as latest')
- Add ComfyUI requirements.txt PR creation with exact template
- Remove time estimates and reorganize for clarity
- Update step numbering and cross-references

christian-byrne committed:
Enable backport automation for already-merged PRs

- Add 'labeled' trigger to backport workflow
- Allow backport automation when needs-backport label is added to merged PRs
- Supports hotfix workflow where labels are added retroactively
- Maintains existing behavior for PRs merged with labels already present

christian-byrne committed:
Prevent duplicate backport triggers with idempotency check

- Add check for existing backport PRs before starting backport process
- Skip backport work if PRs already exist for the same PR number
- Prevents double execution when both 'labeled' and 'closed' events trigger
- Maintains workflow reliability and avoids duplicate backport PRs

christian-byrne committed:
Add smart backport detection to hotfix command

- Check for existing automated backport PRs and their status
- Path A: Skip to version bump if backports already merged
- Path B: Guide user to merge pending backport PRs first
- Path C: Fall back to manual cherry-picking if no/failed automation
- Add clear workflow path documentation for different scenarios

christian-byrne committed:
Add automated fork handling for ComfyUI requirements.txt PRs

- Check if fork exists, create if needed
- Clone fork to local ComfyUI-fork directory
- Create branch, update requirements.txt with sed
- Create PR from fork using gh CLI with exact template format
- Handle both new and existing fork scenarios
- Keep fork directory for future updates

DrJKL commented on .github/workflows/backport.yaml:
```diff
@@ -25,7 +25,26 @@ jobs:
+            echo "Found existing backport PRs:"
+            echo "$EXISTING_BACKPORTS"
+            echo "skip=true" >> $GITHUB_OUTPUT
+            echo "::notice::Backport PRs already exist for PR #${PR_NUMBER}, skipping to avoid duplicates"
```
Nit: Should this be a warning?

DrJKL commented on .github/workflows/backport.yaml:
```diff
@@ -25,7 +25,26 @@ jobs:
+            echo "skip=true" >> $GITHUB_OUTPUT
+            echo "::notice::Backport PRs already exist for PR #${PR_NUMBER}, skipping to avoid duplicates"
+          else
+            echo "skip=false" >> $GITHUB_OUTPUT
```
Insanely nit: want to invert the condition and make this a guard clause? 😁

christian-byrne replied:
done in 68d96772790c981e3817658d956da803514feafc

DrJKL commented on .github/workflows/backport.yaml:
```diff
@@ -25,7 +25,26 @@ jobs:
+          fi
+
       - name: Extract version labels
+        if: steps.check-existing.outputs.skip != 'true'
```
Not for this PR at all, but when a bunch of steps have the same gate applied it makes me think they should be a single [composite action](https://docs.github.com/en/actions/tutorials/create-actions/create-a-composite-action#composite-actions-and-reusable-workflows)

DrJKL reviewed (dismissed):
Only looked at the YAML file, added some small observations, but otherwise 👍🏻

christian-byrne committed:
[style] improve backport workflow logging and structure

- Change ::notice to ::warning for existing backports per @DrJKL's feedback
- Refactor conditional to use guard clause pattern per @DrJKL's suggestion
- Improves readability and follows conventional shell scripting patterns

Addresses @DrJKL's review comments in PR #5271
